src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -69,11 +70,16 @@ public static IEnumerable<QsNamespace> Apply(IEnumerable<QsNamespace> namespaces
                     concreteCallable = globals[currentRequest.originalName].WithFullName(name => currentRequest.concreteName)
```

Comment:
I suggest to use globals.TryGetValue instead of ContainsKey above to avoid looking up the key twice.